internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -128,11 +132,69 @@ private Object doSchemaOperation(String storeName, Function<SchemaData, Object>
     }
   }
 
-  private void mayRegisterAndPopulateMetadataSchema(Store store, SchemaData schemaData) {
+  void maybeRegisterAndPopulateRmdSchema(Store store, SchemaData schemaData) {
     if (store.isActiveActiveReplicationEnabled()) {
       String storeName = store.getName();
-      accessor.subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
-      accessor.getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+      getAccessor().subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
+      getAccessor().getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+    }
+  }
+
+  void maybeRegisterAndPopulateUpdateSchema(Store store, SchemaData schemaData) {
+    if (store.isWriteComputationEnabled()) {
+      String storeName = store.getName();
+      getAccessor().subscribeDerivedSchemaCreationChange(storeName, derivedSchemaChildListener);
+      getAccessor().getAllDerivedSchemas(storeName).forEach(schemaData::addDerivedSchema);
+    }
+  }
+
+  void maybeForceRefreshSchemaDataForSupersetSchemaWithRetry(Store store, SchemaData schemaData, int supersetSchemaId) {
+    if (isSupersetSchemaReadyToServe(store, schemaData, supersetSchemaId)) {
+      return;
+    }
+    RetryUtils.executeWithMaxAttempt(() -> {
+      forceRefreshSchemaData(store, schemaData);
+      if (!isSupersetSchemaReadyToServe(store, schemaData, supersetSchemaId)) {
+        throw new InvalidVeniceSchemaException(
+            "Unable to refresh superset schema id: " + supersetSchemaId + " for store: " + store.getName());
+      }
+    }, 3, Duration.ofNanos(1), Collections.singletonList(InvalidVeniceSchemaException.class));
```

Comment:
duration of nanos ? isnt that too short ?

Reply:
I see copy paste from other places, which is using nanos which is literally CPU level time, please update to 100s  millis or second.

Reply:
I actually think of this, but I don't see why we need sleep between different retry here? It is linked to critical ingestion path as I can see so I am wondering if we need manual delay...